cmake/SundialsSetupCompilers.cmake

```diff
@@ -111,19 +111,19 @@ endif()
 # C settings
 # ===============================================================
 
-set(DOCSTR "The C standard to use (90, 99, 11, 17)")
+set(DOCSTR "The C standard to use (99, 11, 17)")
 sundials_option(CMAKE_C_STANDARD STRING "${DOCSTR}" "99"
-                OPTIONS "90;99;11;17")
+                OPTIONS "99;11;17")
 message(STATUS "C standard set to ${CMAKE_C_STANDARD}")
 
 set(DOCSTR "Enable C compiler specific extensions")
 sundials_option(CMAKE_C_EXTENSIONS BOOL "${DOCSTR}" ON)
 message(STATUS "C extensions set to ${CMAKE_C_EXTENSIONS}")
 
-# Profiling generally requires ISO C99 or newer for __func__ though some
-# compilers define __func__ even with ISO C90.
-if(SUNDIALS_BUILD_WITH_PROFILING AND (CMAKE_C_STANDARD STREQUAL "90"))
-  message(WARNING "SUNDIALS_BUILD_WITH_PROFILING=ON requires __func__, compilation may fail with CMAKE_C_STANDARD=90")
+# Before we do compile checks, ensure warnings are reported as errors
+set(OLD_CMAKE_REQUIRED_FLAGS ${CMAKE_REQUIRED_FLAGS})
+if(NOT ENABLE_WARNINGS_AS_ERRORS)
+  set(CMAKE_REQUIRED_FLAGS "-Werror")
```

Comment:
Will this flag cause any problems on Windows or compilers other than gcc/clang?

Reply:
As per our offline discussions, this would cause a problem so we removed it.